### PR TITLE
[iOS][A11Y] fix hittest with non-SemanticsObject

### PR DESCRIPTION
**Pull request by cyanglaz**
PlatformViewSemanticsContainer did not implement the nativeAccessibility method, leads to a nil object being returned. 

Fixes https://github.com/flutter/flutter/issues/131014

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/SemanticsObject.mm [hunk 1]
```diff
@@ -573,7 +578,13 @@ - (SemanticsObject*)search:(CGPoint)point {
 // IOS 16. Overrides this method to focus the first eligiable semantics
 // object in hit test order.
 - (id)_accessibilityHitTest:(CGPoint)point withEvent:(UIEvent*)event {
-  return [self search:point];
+  id hittest = [self search:point];
+  if (hittest != self) {
```
Comment (hangyujin) [inline shell/platform/darwin/ios/framework/Source/SemanticsObject.mm:582]: can we just check if hittest is a semanticObject here?

Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/SemanticsObject.mm]: sure. done

File: shell/platform/darwin/ios/framework/Source/SemanticsObject.mm [hunk 2]
```diff
@@ -573,7 +578,13 @@ - (SemanticsObject*)search:(CGPoint)point {
 // IOS 16. Overrides this method to focus the first eligiable semantics
 // object in hit test order.
 - (id)_accessibilityHitTest:(CGPoint)point withEvent:(UIEvent*)event {
-  return [self search:point];
+  id hittest = [self search:point];
+  if (![hittest isKindOfClass:[SemanticsObject class]]) {
+    // If hittest result is not a SemanticsObject (e.g. PlatformView),
+    // call the default hittest method to find the hittest result inside the "hittest".
+    return [(NSObject*)hittest _accessibilityHitTest:point withEvent:event];
```
Comment (stuartmorgan) [inline shell/platform/darwin/ios/framework/Source/SemanticsObject.mm:585]: This is not viable, unfortunately.

If we can't implement this method without needing to make a call like this, we'll need to remove the method entirely and find a different workaround for the initial problem it was added to address.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by cyanglaz** [comment]
@stuartmorgan I overthought the problem. The fix is actually a lot simpler than what I had. PTAL

**Pull request merged**